src/modules/mc_pos_control/mc_pos_control_main.cpp

```diff
@@ -787,8 +837,8 @@ MulticopterPositionControl::limit_pos_sp_offset()
 	pos_sp_offs.zero();
 
 	if (_control_mode.flag_control_position_enabled) {
-		pos_sp_offs(0) = (_pos_sp(0) - _pos(0)) / _params.sp_offs_max(0);
-		pos_sp_offs(1) = (_pos_sp(1) - _pos(1)) / _params.sp_offs_max(1);
+		/* Limit maximum setpoint offset in position in body frame */
+		scale_with_body_fixed_gains(pos_sp_offs, _pos_sp - _pos, _params.sp_offs_max.einv(), false);
```

Comment:
What happens if `sp_offs_max` is 0? It looks like the gain is going to be `inv` and the setpoints will end somewhere at infinity as well.